src/server/rest/v1/service/ChargingStationService.ts

```diff
@@ -1221,9 +1226,70 @@ export default class ChargingStationService {
           action: action
         });
       }
+      const tag = await TagStorage.getTagByVisualID(req.tenant.id, filteredRequest.args.visualTagID, { userIDs: [user.id] });
+      if (!tag) {
+        throw new AppError({
+          source: Constants.CENTRAL_SERVER,
+          errorCode: HTTPError.USER_NO_BADGE_ERROR,
+          message: 'The user does not have any badge',
+          module: MODULE_NAME,
+          method: 'handleAction',
+          user: req.user,
+          action: action,
+        });
+      }
+      // Inactive Tag
+      if (!tag.active) {
+        throw new BackendError({
+          source: chargingStation.id,
+          action: action,
+          message: `Tag ID '${tag.id}' is not active`,
+          module: MODULE_NAME, method: 'handleAction',
+          user: tag.user,
+          detailedMessages: { tag }
+        });
+      }
+      // Org component enabled?
+      if (Utils.isTenantComponentActive(req.tenant, TenantComponents.ORGANIZATION)) {
+        let foundSiteArea = true;
+        // Site Area -----------------------------------------------
+        if (!chargingStation.siteAreaID) {
+          foundSiteArea = false;
+        } else if (!chargingStation.siteArea) {
+          chargingStation.siteArea = await SiteAreaStorage.getSiteArea(
+            req.tenant.id, chargingStation.siteAreaID, { withSite: true });
+          if (!chargingStation.siteArea) {
+            foundSiteArea = false;
+          }
+        }
+        // Site is mandatory
+        if (!foundSiteArea) {
+          // Reject Site Not Found
+          throw new BackendError({
+            source: chargingStation.id,
+            action: action,
+            module: MODULE_NAME, method: 'handleAction',
+            message: `Charging Station '${chargingStation.id}' is not assigned to a Site Area!`,
+            detailedMessages: { chargingStation }
+          });
+        }
+        // Site -----------------------------------------------------
+        chargingStation.siteArea.site = chargingStation.siteArea.site ??
```

Comment:
You can retrieve the charging station with the Site also (chargingStation.site)

Reply:
It's not needed anymore, I'v replaced it with Authorization.isChargingStationValidInOrganization()